src/character_body.cpp

```diff
@@ -464,7 +465,8 @@ void Character::update_bodytemp()
     const bool has_bark = has_flag( json_flag_BARKY );
     const bool has_sleep = has_effect( effect_sleep );
     const bool has_sleep_state = has_sleep || in_sleep_state();
-    const bool heat_immune = has_flag( json_flag_HEAT_IMMUNE );
+    const bool fire_immune = has_flag( json_flag_FIRE_IMMUNE ) || worn_with_flag( flag_FIRE_IMMUNE );
+    const bool heat_immune = has_flag( json_flag_HEAT_IMMUNE ) || worn_with_flag( flag_HEAT_IMMUNE );
```

Comment:
look around line 520 below:
`const units::temperature_delta h_radiation = get_heat_radiation( pos() );`
Just after that you want to adjust down h_radiation based on how good the armor is at reflecting thermal energy.
Likewise player_local_temp is acquired up on line 452, adjust that based on clothing's ability to reject heat.